### PR TITLE
fixing private IPs for openstack

### PR DESCRIPTION
**Pull request by nevins-b**
This changes fixes and issue where data.private_ips was not being set if public_ips where provided. It resolves issue [199](https://github.com/saltstack/salt-cloud/issues/199).

### CODE REVIEW

File: saltcloud/clouds/openstack.py [hunk 1]
```diff
@@ -412,6 +412,8 @@ def create(vm_):
 
         if running and public:
             data.public_ips = public
+            if private:
```
Comment (s0undt3ch) [inline saltcloud/clouds/openstack.py]: Shouldn't this code be:

``` python
if running and public:
    data.public_ips = public
if running and private:
    data.private_ips = private
```

Comment (nevins-b) [inline saltcloud/clouds/openstack.py]: I don't think so, I think if anything you may want it to look at ssh_interface(vm_) and see if it should continue waiting for private IPs once it has public ones. I would think 90% of the time people will want to code to return after it has the public IPs.

Comment (s0undt3ch) [inline saltcloud/clouds/openstack.py]: My change does not add to the wait, I think.

Comment (s0undt3ch) [inline saltcloud/clouds/openstack.py]: Ah, sorry, I removed the `not_ready = False`

Comment (s0undt3ch) [inline saltcloud/clouds/openstack.py]:
``` python
if running and public:
    data.public_ips = public
    not_ready = False
if running and private:
    data.private_ips = private
```

?

Comment (s0undt3ch) [inline saltcloud/clouds/openstack.py]: Like the above, even if public ips is not know already, and private is, it's set. Break out of the loop when the public ip is found?

Comment (s0undt3ch) [inline saltcloud/clouds/openstack.py]:
``` python
if running and public:
    data.public_ips = public
    not_ready = False
if running and private:
    data.private_ips = private
    if ssh_interface(vm_) == 'private_ips':
        not_ready = False
```

This should be it.

Comment (s0undt3ch) [inline saltcloud/clouds/openstack.py]: Want to update the code so you can still own the change wrights? Just push the new change to your repository and it should show up here.

**Comment by nevins-b** [comment]
Hmm I think what we actually will want is:

``` python
if running and private:
    data.private_ips = private
        if ssh_interface(vm_) == 'private_ips':
            not_ready = False

 if running and public:
     data.public_ips = public
         if ssh_interface(vm_) != 'private_ips':
             not_ready = False
```

This will set both private and public IPs as well as waiting for private to be set if ssh_interface(vm_) == 'private_ips'

**Comment by s0undt3ch** [comment]
Thanks @nevins-b!

**Pull request merged**